# PHYS 6260: Homework 1, Mi Do

In [42]:
import numpy as np

## Problem 1: Altitude of a satellite

A satellite is to be launched into a circular orbit around the Earth so that it orbits the planet once every T seconds. Treating the Earth as a perfect sphere with its center of mass in the middle, it can shown with classical mechanics that the altitude h above the Earth's surface must be 

\begin{equation*}
h = \left(\frac{GMT^2}{4\pi^2}\right)^{(1/3)}-R
\\\text{with } G = 6.67 \times 10^{-11}m^3kg^{-1}s^{-2}\text{ is Newton's gravitational constant, }M = 5.97 \times 1024 \text{ kg is the Earth mass, and } R = 6371 \text{ km is its radius.}
\end{equation*}

In [43]:
#Constant

G = 6.67*10**(-11) #Newton's gravitational constant
M = 5.97*10**24    #Earth mass
R = 6371*1000      #Earth radius

In [47]:
# function accept period in seconds as variable to return corresponding altitude

def altitude(period):
    h = ((G*M*period**2)/(4*np.pi**2))**(1/3)-R
    print("With a period of %.2f seconds, the altitudes will be %.2f m or %.2e km" % (period,h,h))
    return h

### Part (a)

In [48]:
#ask user to input desire value of period in seconds (T), 
#using function altitude(T) to calculate and print out the corresponding altitude in meters

prompt1 = input("What is the value of T in seconds? ") 
t = float(prompt1)
h1 = altitude(t)

What is the value of T in seconds? 9000
With a period of 9000.00 seconds, the altitudes will be 2977497.49 m or 2.98e+06 km


### Part (b)

In [49]:
#using the function altitue(period) to calculate the altitudes of satellites that orbit the Earth:

#a) once a day
t1 = 24*60*60 #changing to seconds
print("One day is equal to %.2f seconds." % t1)
altitude1 = altitude(t1)

#b) once every 90 minutes
t2 = 90*60 #changing to seconds
print("90 minutes is equal to %.2f seconds." % t2)
altitude2 = altitude(t2)

#c) once every 45 minutes
t3 = 45*60 #changing to seconds
print("45 minutes is equal to %.2f seconds." % t3)
altitude3 = altitude(t3)

One day is equal to 86400.00 seconds.
With a period of 86400.00 seconds, the altitudes will be 35855910.18 m or 3.59e+07 km
90 minutes is equal to 5400.00 seconds.
With a period of 5400.00 seconds, the altitudes will be 279321.63 m or 2.79e+05 km
45 minutes is equal to 2700.00 seconds.
With a period of 2700.00 seconds, the altitudes will be -2181559.90 m or -2.18e+06 km


One day is equal to 86400.00 seconds.
With a period of 86400.00 seconds, the altitudes will be 35855910.18 m or 3.59e+07 km
90 minutes is equal to 5400.00 seconds.
With a period of 5400.00 seconds, the altitudes will be 279321.63 m or 2.79e+05 km
45 minutes is equal to 2700.00 seconds.
With a period of 2700.00 seconds, the altitudes will be -2181559.90 m or -2.18e+06 km

With a period of 45 minutes, we get the negative altitudes. This means the satellite orbits within inside the Earth, which seems impossible!

### Part (c)

In [50]:
#calculate the difference in altitude between a period of 23.93 hours and 24 hours
t4 = (24-23.93)*60*60
altidude4 = altitude(t4)

With a period of 252.00 seconds, the altitudes will be -5508986.96 m or -5.51e+06 km


The sidereal day is about 4 minutes shorter than 24 hours, because while the Earth rotates around its axis, it also orbits around the Sun, so, in order to reach the same previous Earth - Sun position, the Earth needs to rotate a liitle longer compared to reach the same previous Earth - "fixed" star position. From the calculation above, we can see that this short amount of time makes a huge difference in the altitude of the satellite, about 5.5 km. 

## Problem 2: Special relativity

A spaceship travels from Earth in a straight line at a relativistic speed v to another planet x light-years away. Time dilation is a special relativistic effect. An observer with some speed b relatice to another observer will experience time progressing at a slower rate. Let $\Delta t'$ be the time interval for the moving frame, and $\Delta t$ be the rest frame. They are related through the equation below with c is the speed of light:

\begin{equation*}
\Delta t' = \Delta t \left(1-\frac{v^2}{c^2}\right)^{1/2} 
\end{equation*}

Since we consider v as a fraction of the speed of light c, the equation above becomes

\begin{equation*}
\Delta t' = \Delta t (1-a^2)^{1/2} \text{ with v = ac} 
\end{equation*}

In [72]:
#function accepts a and x as variables 
#and print out the time in years that spaceship takes to reach its destination 
#in the rest frame of an observer on Earth and in a passenger on board the ship

def time(x,a):
    t = x/a
    tprime = t*(1-a**2)**(1/2)
    print("\n The time that the spaceship takes to reach a planet %i light-years away with v = %.2fc in rest frame of an observer on Earth is %.2f years, and as perceived by a passenger on board the ship is %.2f years." % (x,a,t,tprime))
    return t

In [73]:
#ask user to input desire value of x light years and the speed v as a fraction of the speed of light c (a)
#using the function time(x,a) to calculate and print out the time in years in the rest frame and moving frame.

prompt2 = input("What is the value of x in light years? ") 
x1 = float(prompt2)
prompt3 = input("What is the value of the speed v as a fraction of the speed of light c? ")
a1 = float(prompt3)
t1 = time(x1,a1)

What is the value of x in light years? 50
What is the value of the speed v as a fraction of the speed of light c? 0.5

 The time that the spaceship takes to reach a planet 50 light-years away with v = 0.50c in rest frame of an observer on Earth is 100.00 years, and as perceived by a passenger on board the ship is 86.60 years.


In [74]:
#using the funtion time(x,a) to calculate the time traveled in both frames 
#for a planet 10 light-years away with v = 0.99c or a = 0.99 and x = 10

x2 = 10
a2 = 0.99
t2 = time(x2,a2)


 The time that the spaceship takes to reach a planet 10 light-years away with v = 0.99c in rest frame of an observer on Earth is 10.10 years, and as perceived by a passenger on board the ship is 1.42 years.


The time that the spaceship takes to reach a planet 10 light-years away with v = 0.99c in rest frame of an observer on Earth is 10.10 years, and as perceived by a passenger on board the ship is 1.42 years.

## Problem 3: The semi-empirical mass formula

In nuclear physics, the semi-empirical mass formula calculates the approximate nuclear binding energy B of an atomic nucleus with atomic number Z and mass number A. The formula reads as:

\begin{equation*}
B = a_1A - a_2A^{2/3} - a_3\frac{Z^2}{A^{1/3}}\ - a_4\frac{(A-2Z)^2}{A}\ + \frac{a_5}{A^{1/2}}\
\end{equation*}


where (in units of millions of electron volts, MeV), the constants $a_1$ = 15.67, $a_2$ = 17.23, $a_3$ = 0.75, $a_4$ = 93.2, and

\begin{equation*}
a_5 =\begin{cases}
0 & \quad \text{if }  \text{A is odd,}\\
12.0 & \quad \text{if }  \text{A and Z are both even,}\\
-12.0 & \quad \text{if }  \text{A is even and Z is odd.}
 \end{cases}
\end{equation*}

In [76]:
#constant
a1 = 15.67
a2 = 17.23
a3 = 0.75
a4 = 93.2

In [109]:
#function accepts A and Z as variables 
#then check if A, Z are odd or even to choose an appropriate constant a5
#then using the given function to calculate the approximate nuclear binding energy B
#and binding energy per nucleon 
#finally, print out the answer and return B

def binding(A,Z):
    #check if A, Z are odd or even to choose to a constant a5
    if (A&1) == 1:
        a5 = 0
    else:
        if (Z&1) == 1:
            a5 = -12.0
        else: 
            a5 = 12.0
            
    #calculate B and B_nucleon
    B = a1*A - a2*A**(2/3) - a3*Z**2/A**(1/3) - a4*(A-2*Z)**2/A + a5/A**(1/2)
    B_nucleon = B/A
    
    print("Given A=%i & Z=%i, the binding energy B for the corresponding atom is %.2f MeV, and the binding energy per nucleon is %.2f MeV" % (A,Z,B,B_nucleon))
    return B

In [110]:
#ask user to input desire value of mass number A and atomic number Z
#using the function binding(A,Z) to calculate 
#and print out the binding energy B and the binding energy per nucleon B_nucleon

print("Consider an atomic nucleus:")
prompt4 = input("What is the value of mass number A? ") 
A1 = int(prompt4)
prompt5 = input("What is the value of atomic number Z? ")
Z1 = int(prompt5)
B1 = binding(A1,Z1)

Consider an atomic nucleus:
What is the value of mass number A? 28
What is the value of atomic number Z? 31
Given A=28 & Z=31, the binding energy B for the corresponding atom is -3807.57 MeV, and the binding energy per nucleon is -135.98 MeV


In [112]:
A2 = 58
Z2 = 28
B2 = binding(A2,Z2)

Given A=58 & Z=28, the binding energy B for the corresponding atom is 493.94 MeV, and the binding energy per nucleon is 8.52 MeV


Given A=58 & Z=28, the binding energy B for the corresponding atom is 493.94 MeV, and the binding energy per nucleon is 8.52 MeV